cnspec panics when running in GitHub Actions

**Describe the bug**
I tried to run the image scan GitHub action with cnspec inside a container.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x637d5b0]

goroutine 1 [running]:
go.mondoo.com/cnspec/policy/scan.(*LocalScanner).distributeJob(0xc0007fc780, 0xc001319bd0, {0x86ee748?, 0xc0011d4f60}, {{0x0, 0x0}, {0xc00116af80, 0x35}, {0xc0000c2120, 0x19}, ...})
	/home/benr/actions-runner/_work/cnspec/cnspec/policy/scan/local_scanner.go:180 +0xf50
go.mondoo.com/cnspec/policy/scan.(*LocalScanner).Run(0xc0007fc780, {0x86ee748?, 0xc0011d4f00?}, 0xc0009bfbd0)
	/home/benr/actions-runner/_work/cnspec/cnspec/policy/scan/local_scanner.go:100 +0x12c
go.mondoo.com/cnspec/apps/cnspec/cmd.RunScan(0xc0010f7e00)
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/scan.go:515 +0x317
go.mondoo.com/cnspec/apps/cnspec/cmd.glob..func18(0x0?, {0xc00031fb90?, 0x0?, 0x0?}, 0x0?, 0x0?)
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/scan.go:347 +0xc5
go.mondoo.com/cnquery/apps/cnquery/cmd/builder.dockerImageProviderCmd.func1(0xc000f4d200?, {0xc00031fb90?, 0x7?, 0x7?})
	/home/benr/go/pkg/mod/go.mondoo.com/cnquery@v0.0.0-20221031224435-72f91fc1103c/apps/cnquery/cmd/builder/builder.go:387 +0x2e
github.com/spf13/cobra.(*Command).execute(0xc000f4d200, {0xc00031fab0, 0x7, 0x7})
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:920 +0x847
github.com/spf13/cobra.(*Command).ExecuteC(0xbb18a40)
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:1044 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/benr/go/pkg/mod/github.com/spf13/cobra@v1.6.1/command.go:[96](https://github.com/mondoohq/actions/actions/runs/3376068206/jobs/5603368838#step:6:97)8
go.mondoo.com/cnspec/apps/cnspec/cmd.Execute()
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cmd/root.go:70 +0x25
main.main()
	/home/benr/actions-runner/_work/cnspec/cnspec/apps/cnspec/cnspec.go:6 +0x17
```

**To Reproduce**
Steps to reproduce the behavior:
1. Start the workflow in this PR: https://github.com/mondoohq/actions/pull/40
2. Wait for the image scan to start
3. Wait for it to crash
4. See error

**Expected behavior**
Scan should run without error.

**Screenshots**

**Desktop (please complete the following information):**
cnspec latest image:
```
Digest: sha256:f9350303514e8b16af624304215318a2f9ef4ff74b28a64cdc9e6e850f6393a5
Status: Downloaded newer image for mondoo/cnspec:latest
```

**Additional context**
Link to a failed run:
https://github.com/mondoohq/actions/actions/runs/3376068206/jobs/5603368838


From what I currently know, it is a mismatch between the platform-id and the platformMrn:
```
→ syncing assets
DBG got assets details assets=1
DBG asset mapping asset=//assets.api.mondoo.app/spaces/mondoo-github-actions/assets/2Gz67q8Woe5MsAztNXK8RwwC8hf platform-mrn=//platformid.api.mondoo.app/runtime/cicd/actions.github.com/mondoohq-actions/ref/refs-pull-40-merge/run/3376068206/job/docker-build-scan-push/action/__self/hash/3e84d63b07343b5fa4fbf5dbf6fbf41fefd7aa76b4c06ebcfeb1ebde31d072ff
DBG update asset asset=index.docker.io/library/ubuntu@7cfe75438fc7 platform-ids=["//platformid.api.mondoo.app/runtime/docker/images/7cfe75438fc77c9d7235ae502bf229b15ca86647ac01c844b272b56326d56184"]
```

--- a/.vscode/launch.json
+++ b/.vscode/launch.json
@@ -55,6 +55,67 @@
       ],
       "env": {
       }
+    },
+    {
+      "name": "Scan container image (CICD fake)",
+      "type": "go",
+      "request": "launch",
+      "program": "${workspaceRoot}/apps/cnspec/cnspec.go",
+      "cwd": "${workspaceRoot}/",
+      "args": [
+        "scan",
+        "docker",
+        "image",
+        "ubuntu:22.04",
+        "--log-level",
+        "debug",
+        "--config",
+        "/home/christian/workspace/mondoo/gitlab.com/mondoo/demo.agent.credentials.json"
+      ],
+      "env": {
+        "GITHUB_ACTION": "__self",
+        "GITHUB_ACTIONS": "true",
+        "GITHUB_ACTION_PATH": "",
+        "GITHUB_ACTOR": "czunker",
+        "GITHUB_RUN_ID": "3376068206",
+        "GITHUB_RUN_ATTEMPT": "1",
+        "GITHUB_RUN_NUMBER": "107",
+        "GITHUB_REPOSITORY": "mondoohq/actions",
+        "GITHUB_REPOSITORY_OWNER": "mondoohq",
+        "GITHUB_EVENT_NAME": "pull_request",
+        "GITHUB_EVENT_PATH": "/github/workflow/event.json",
+        "GITHUB_WORKSPACE": "/github/workspace",
+        "GITHUB_SHA": "7a9cedc90fc259c3389515b2a0525383240d7c5a",
+        "GITHUB_REF": "refs/pull/40/merge",
+        "GITHUB_REF_NAME": "40/merge",
+        "GITHUB_HEAD_REF": "christian/switch_to_container_image",
+        "GITHUB_BASE_REF": "main",
+        "GITHUB_REF_TYPE": "branch",
+        "GITHUB_JOB": "docker-build-scan-push",
+        "GITHUB_SERVER_URL": "https://github.com",
+        "GITHUB_WORKFLOW": "Docker Image Scanning",
+        "GITHUB_REF_PROTECTED": "false",
+        "RUNNER_ARCH": "X64",
+        "RUNNER_NAME": "GitHub Actions 3",
+        "RUNNER_OS": "Linux"
+      }
+    },
+    {
+      "name": "Scan container image",
+      "type": "go",
+      "request": "launch",
+      "program": "${workspaceRoot}/apps/cnspec/cnspec.go",
+      "cwd": "${workspaceRoot}/",
+      "args": [
+        "scan",
+        "docker",
+        "image",
+        "ubuntu:22.04",
+        "--log-level",
+        "debug",
+        "--config",
+        "/home/christian/workspace/mondoo/gitlab.com/mondoo/demo.agent.credentials.json"
+      ],
     }
   ]
 }

--- a/policy/scan/local_scanner.go
+++ b/policy/scan/local_scanner.go
@@ -169,16 +169,21 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstreamConf
 		log.Debug().Int("assets", len(resp.Details)).Msg("got assets details")
 		platformAssetMapping := make(map[string]*policy.SynchronizeAssetsRespAssetDetail)
 		for i := range resp.Details {
-			log.Debug().Str("platform-mrn", resp.Details[i].PlatformMrn).Str("asset", resp.Details[i].AssetMrn).Msg("asset mapping")
-			platformAssetMapping[resp.Details[i].PlatformMrn] = resp.Details[i]
+			log.Debug().Str("platform-mrn", i).Str("asset", resp.Details[i].AssetMrn).Msg("asset mapping")
+			platformAssetMapping[i] = resp.Details[i]
 		}
 
 		// attach the asset details to the assets list
 		for i := range assetList {
 			log.Debug().Str("asset", assetList[i].Name).Strs("platform-ids", assetList[i].PlatformIds).Msg("update asset")
-			platformMrn := assetList[i].PlatformIds[0]
-			assetList[i].Mrn = platformAssetMapping[platformMrn].AssetMrn
-			assetList[i].Url = platformAssetMapping[platformMrn].Url
+			platformId := assetList[i].PlatformIds[0]
+			if err != nil {
+				return nil, false, errors.Wrap(err, "failed to generate a platform MRN")
+			}
+			assetList[i].Mrn = platformAssetMapping[platformId].AssetMrn
+			assetList[i].Url = platformAssetMapping[platformId].Url
+			// only needed for CI/CD, can be removed when fleet and CI/CD have the same platform MRN
+			assetList[i].PlatformIds = []string{platformAssetMapping[platformId].PlatformMrn}
 		}
 	} else {
 		// ensure we have non-empty asset MRNs

--- a/policy/scan/local_scanner.go
+++ b/policy/scan/local_scanner.go
@@ -177,9 +177,6 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstreamConf
 		for i := range assetList {
 			log.Debug().Str("asset", assetList[i].Name).Strs("platform-ids", assetList[i].PlatformIds).Msg("update asset")
 			platformId := assetList[i].PlatformIds[0]
-			if err != nil {
-				return nil, false, errors.Wrap(err, "failed to generate a platform MRN")
-			}
 			assetList[i].Mrn = platformAssetMapping[platformId].AssetMrn
 			assetList[i].Url = platformAssetMapping[platformId].Url
 			// only needed for CI/CD, can be removed when fleet and CI/CD have the same platform MRN

